crates/red_knot_python_semantic/src/types.rs

```diff
@@ -416,21 +416,26 @@ impl<'db> Type<'db> {
             (Type::Never, _) => true,
             (_, Type::Never) => false,
             (Type::BooleanLiteral(_), Type::Instance(class))
-                if class.is_known(db, KnownClass::Bool) =>
+                if matches!(class.known(db), Some(KnownClass::Bool | KnownClass::Object)) =>
```

Comment:
Another way of doing this would be to move these branches higher up so that they come immediately below the `Type::Never` branches, before any of the `Literal` branches:

https://github.com/astral-sh/ruff/blob/cd6c9371943db7d10fc5c449eccfb2700c485cd2/crates/red_knot_python_semantic/src/types.rs#L445-L446

Which might be slightly less efficient, but also slightly less error-prone? Since we won't have to remember to account for `object` in all the other branches if we deal with them upfront at the top of the `match`

Reply:
That also works, yes. I thought about it.

> Which might be slightly less efficient

That's why I didn't implement it this way. Happy to change it though, if you think it helps. On the other hand, if we add new type variants, we would probably check what other branches do.. but then again, your solution would be less code.

Reply:
I don't mind too much; happy to defer to what you think is best. I'm sure you have better instincts than me for what will be best in terms of performance ;)

Reply:
Let's not trust any instincts when it comes to performance. I like your version better — changed.